pkg/remote/trans/nphttp2/stream.go

```diff
@@ -95,7 +95,11 @@ func (s *stream) RecvMsg(m interface{}) error {
 	ri := rpcinfo.GetRPCInfo(s.ctx)
 
 	msg := remote.NewMessage(m, s.svcInfo, ri, remote.Stream, remote.Client)
-	msg.SetProtocolInfo(remote.NewProtocolInfo(ri.Config().TransportProtocol(), s.svcInfo.PayloadCodec))
+	if s.svcInfo == nil {
+		msg.SetProtocolInfo(remote.NewProtocolInfo(ri.Config().TransportProtocol(), serviceinfo.Protobuf))
```

Comment:
@HeyJavaBean In the previous logic, `s.svcInfo.PayloadCodec` would not cause an error because `svcInfo` is never nil. But in this logic, `s.svcInfo` can become nil, so I wrote like this. This certainly looks a bit weird though...

Reply:
@HeyJavaBean I check svcInfo == nil because I thought that svcInfo could be nil when the service is an unknown service and unknownServiceHandler is called. But I'm not sure if` RecvMsg` can be called inside the unknown service handler or not.

Reply:
But you change s.svcInfo.PayloadCodec -> serviceinfo.Protobuf

Reply:
@HeyJavaBean yes... I don't know what to put as PayloadCodec when we can't get it from s.svcInfo...

Reply:
@HeyJavaBean I'll set the default svcInfo when svc == nil, then there's no need to add this weird logic anymore.

Reply:
@HeyJavaBean I'll set the default service when svc == nil because svcInfo is different from the service client calls in the original logic. Then we don't need to add this weird logic.